### PR TITLE
adding tools to master in prep for go mod.

### PR DESCRIPTION
**Pull request by n3wscott**

### CODE REVIEW

File: hack/tools.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tools
+
+// This package imports things required by this repository, to force `go mod` to see them as dependencies
+import (
+	_ "k8s.io/code-generator"
```
Comment (mattmoor) [inline hack/tools.go:23]: I thought you needed to list individual code-generators, do you not?

Should this list the main package for our code-generator as well?

Comment (chizhg) [reply hack/tools.go:23]: This is how k8s/sample-controller does - https://github.com/kubernetes/sample-controller/blob/master/hack/tools.go#L22

Comment (n3wscott) [reply hack/tools.go:23]: I don't think this is working, I do think you have to include the individual code-generators

Comment (n3wscott) [reply hack/tools.go:23]: it turns out you need both.

**Comment by vaikas** [comment]
shouldn't the Gopkg.* files be totally gone?

**Comment by n3wscott** [comment]
Not migrating to go mod here, just prep for downstream to include the hack dir in downstream vendor dirs.

File: hack/tools.go (hunk 1 above)
Comment (chizhg) [reply hack/tools.go:23]: Hmm, they are already imported in https://github.com/kubernetes/code-generator/blob/master/tools.go#L24-L34. I think it should be enough for Go modules?

**Pull request merged**